workspaces/arborist/lib/override-set.js

```diff
@@ -44,6 +44,43 @@ class OverrideSet {
     }
   }
 
+  childrenAreEqual (other) {
```

Comment:
I implemented childrenAreEqual in a way to avoid recursion...
```js
const LOOP_SENTINEL = 1_000_000
```
then 
```js
childrenAreEqual(otherOverrideSet: OverrideSetClass): boolean {
  const queue = [[this, otherOverrideSet]]
  let pos = 0
  let { length: queueLength } = queue
  while (pos < queueLength) {
    if (pos === LOOP_SENTINEL) {
      throw new Error('Detected infinite loop while comparing override sets')
    }
    const { 0: currSet, 1: currOtherSet } = queue[pos++]
    const { children } = currSet
    const { children: otherChildren } = currOtherSet
    if (children.size !== otherChildren.size) {
      return false
    }
    for (const key of children.keys()) {
      if (!otherChildren.has(key)) {
        return false
      }
      const child = children.get(key)
      const otherChild = otherChildren.get(key)
      if (child!.value !== otherChild!.value) {
        return false
      }
      queue[queueLength++] = [child, otherChild]
    }
  }
  return true
}
```

Reply:
I think the recursion is more readable, and that stack exhaustion isn't a real threat.
These are override sets. So the depth here it the depth of the overrides section in the package.json. I don't think we'll see depth = 5 in a realistic use case, and if someone on purpose feeds npm with an overrides section with depth = 100000 then he's only hurting himself?